src/main/scala/edu/ie3/simona/model/participant2/HpModel.scala

```diff
@@ -0,0 +1,493 @@
+      case Some(threshold) => Some(threshold.tick)
+      case None            => None
+    }
+
+    (operatingPoint, nextTick)
+  }
+
+  override def determineOperatingPoint(
+      state: HpState,
+      setPower: Power,
+  ): (HpOperatingPoint, OperationChangeIndicator) = {
+
+    val (newActivePowerHp, qDotIntoGrid) =
+      nextOperatingPoint(state, Some(setPower))
+
+    /* Push thermal energy to the thermal grid and get its updated state in return */
+    val (updateState, maybeThreshold) = thermalGrid.updateState(
+      state.tick,
+      state,
+      newActivePowerHp > zeroKW,
+      qDotIntoGrid,
+      state.thermalDemands,
+    )
+
+    val operatingPoint =
+      HpOperatingPoint(
+        newActivePowerHp,
+        Some(
+          ThermalOpWrapper(
+            qDotIntoGrid,
+            updateState.houseState.map(_.qDot).getOrElse(zeroKW),
+            updateState.storageState.map(_.qDot).getOrElse(zeroKW),
+          )
+        ),
+      )
+
+    val nextTick = maybeThreshold match {
+      case Some(threshold) => Some(threshold.tick)
+      case None            => None
+    }
+
+    (
+      operatingPoint,
+      OperationChangeIndicator(
+        changesAtNextActivation = true,
+        changesAtTick = nextTick,
+      ),
+    )
+  }
+}
+
+object HpModel {
+
+  final case class HpOperatingPoint(
+      override val activePower: Power,
+      override val thermalOps: Option[ThermalOpWrapper],
+  ) extends OperatingPoint {
+    override val reactivePower: Option[ReactivePower] = None
+  }
+
+  object HpOperatingPoint {
+    def zero: HpOperatingPoint =
+      HpOperatingPoint(zeroKW, Some(ThermalOpWrapper(zeroKW, zeroKW, zeroKW)))
+  }
+
+  /** Wraps the thermal powers of the [[HpOperationPoint]].
+    *
+    * @param qDotHp
+    *   The thermal power output of the heat pump.
+    * @param qDotHouse
+    *   The thermal power input of the [[ThermalHouse]].
+    * @param qDotHeatStorage
+    *   The thermal power input of the [[ThermalStorage]].
+    */
+  final case class ThermalOpWrapper private (
+      qDotHp: Power,
+      qDotHouse: Power,
+      qDotHeatStorage: Power,
+  )
+
+  /** Holds all relevant data for a hp model calculation.
+    *
+    * @param tick
+    *   The current tick.
+    * @param ambientTemperature
+    *   The actual outside temperature.
+    * @param thermalGridState
+    *   The applicable state of the [[ThermalGrid]].
+    * @param lastHpOperatingPoint
+    *   The last [[HpOperatingPoint]] of the heat pump.
+    * @param lastAmbientTemperature
+    *   The outside temperature at the lastState.
+    * @param thermalDemands
+    *   The actual thermal demands of the thermal grid elements (house,
+    *   storage).
+    */
+  final case class HpState(
+      override val tick: Long,
+      ambientTemperature: Temperature,
+      thermalGridState: ThermalGridState,
+      lastHpOperatingPoint: HpOperatingPoint,
+      lastStateAmbientTemperature: Temperature,
```

Comment:
`lastStateAmbientTemperature` should not be needed anymore. In the process of calculating the current state, you should still have acceess to the `ambientTemperature` of the previous state. After calculating the current state, you shouldn't need the previous ambient temp. anymore, as far as I can tell.

Reply:
I don't agree. You're right in principle, but we would be forcing this concept into ThermalHouse. Maybe it's worth it, maybe not. But I think it should be done in a separate PR. In ThermalHouse.updateState() we actually need the lastAmbientTemp to determine the current state. And the ambientTemp to determine the threshold.

If we split this into two methods we could do what you're suggesting.

Reply:
this would be solved by #1329